src/main/clojure/aerospike_clj/client.clj

```diff
@@ -172,390 +105,367 @@
-         op-future (p/deferred)
-         start-time (System/nanoTime)]
-     (.info ^AerospikeClient client
-            ^EventLoop (.next ^EventLoops (:el db))
-            (AsyncInfoListener. op-future)
-            ^InfoPolicy (:policy conf (.infoPolicyDefault ^AerospikeClient client))
-            node
-            (into-array String info-commands))
-     (register-events op-future db "info" nil start-time))))
-
-(defn get-nodes [db]
-  (.getNodes (get-client db)))
-
-;; metrics
-(defn get-cluster-stats
-  "For each client, return a vector of [metric-name metric-val] 2-tuples.
-  The metric name is a dot separated string that should be convenient for
-  reporting to statsd/graphite. All values are gauges."
-  [db]
-  (let [clients (get-all-clients db)]
-    (->> clients
-         (mapv #(metrics/construct-cluster-metrics (.getClusterStats ^AerospikeClient %)))
-         (mapv metrics/cluster-metrics->dotted))))
-
-;; health
-
-(defn healthy?
-  "Returns true iff the cluster is reachable and can take reads and writes.
-  Uses __health-check set to avoid data collisions. `operation-timeout-ms` is
-  for total timeout of reads (default is 1s) including 2 retries so a small
-  over estimation is advised to avoid false negatives."
-  ([db]
-   (healthy? db 1000))
-  ([db operation-timeout-ms]
-   (let [read-policy (let [p (.readPolicyDefault ^AerospikeClient (get-client db ""))]
-                       (set! (.totalTimeout p) operation-timeout-ms)
-                       p)
-         k (str "__health__" (rand-int Integer/MAX_VALUE))
-         v (rand-int Integer/MAX_VALUE)
-         ttl (min 1 (int (/ operation-timeout-ms 1000)))
-         set-name "__health-check"]
-     (try
-       @(put db k set-name v ttl)
-       (= v
-          @(get-single db k set-name {:transcoder :payload
-                                      :policy read-policy}))
-       (catch Exception _ex
-         false)))))
-
-;; etc
+            ^WritePolicy policy
+            ^Key (pt/create-key index dbns set-name)
+            ^"[Lcom.aerospike.client.Bin;" (utils/v->array Bin (mapv set-bin-as-null bin-names)))
+      (register-events op-future client-events :write index start-time)))
+
+  pt/AerospikeBatchOps
+  (get-batch [this batch-reads]
+    (pt/get-batch this batch-reads {}))
+
+  (get-batch [_this batch-reads conf]
+    (let [op-future       (p/deferred)
+          start-time      (System/nanoTime)
+          batch-reads-arr (ArrayList. ^Collection (mapv #(map->batch-read % dbns) batch-reads))]
+      (.get ^AerospikeClient client
+            ^EventLoop (.next ^EventLoops el)
+            (AsyncBatchListListener. op-future)
+            ^BatchPolicy (:policy conf)
+            ^List batch-reads-arr)
+      (let [d (p/chain op-future
+                       #(mapv batch-read->map %)
+                       (:transcoder conf identity))]
+        (register-events d client-events :read-batch nil start-time))))
+
+  (get-multiple [this indices sets]
```

Comment:
Didn't we deprecate it?

Reply:
Not yet. This might be a good time to do so.

Reply:
Deprecated.